docker/cgc_metadata_to_qiime_mapping_file.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, Gregory Poore.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# ----------------------------------------------------------------------------
+import numpy as np
+import pandas as pd
+import sevenbridges as sbg
+import glob, sys, getopt, os, json, unicodedata, argparse, time
+from pandas.io.json import json_normalize
+from tqdm import tqdm # progressbar
+from tqdm import trange
+
+parser = argparse.ArgumentParser(description="""
```

Comment:
Doesn't need to be click but it is very clean and very simple especially
for nested command structures. Argparse is also, unfortunately, not POSIX
compliant

On Aug 24, 2016 12:33, "Greg Poore" <notifications@github.com> wrote:

> In docker/cgc_metadata_to_qiime_mapping_file.py
> <https://github.com/biocore/tcga/pull/1#discussion_r76123428>:
>
> > +#!/usr/bin/env python3
> > +# ----------------------------------------------------------------------------
> > +# Copyright (c) 2016--, Gregory Poore.
> > +#
> > +# Distributed under the terms of the Modified BSD License.
> > +#
> > +# ----------------------------------------------------------------------------
> > +import numpy as np
> > +import pandas as pd
> > +import sevenbridges as sbg
> > +import glob, sys, getopt, os, json, unicodedata, argparse, time
> > +from pandas.io.json import json_normalize
> > +from tqdm import tqdm # progressbar
> > +from tqdm import trange
> > +
> > +parser = argparse.ArgumentParser(description="""
>
> argparse was mostly chosen out of convenience but also because various
> sources said it was the recommended CLI parsing module in the Python
> standard library. If it needs to be ported to click, it can be done.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/tcga/pull/1/files/eab95f73fb3f0519e275159d1c0e1481762b94ef..dda326f55f4f59a7698f267f00698e5934ad1830#r76123428>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAc8siN348YRun_0uEg2CLeXkeNRcKaVks5qjJyKgaJpZM4JgsdH>
> .
>